### PR TITLE
View product

### PR DESCRIPTION
**Pull request by illright**

### CODE REVIEW

**Comment by illright** [comment]
@illright What could speed things up is this review @aabounegm @VanishMax

File: src/routes/product/[id([0-9]+)]/index.svelte [hunk 1]
```diff
@@ -0,0 +1,210 @@
+<script context="module">
```
Comment (VanishMax) [inline src/routes/product/[id([0-9]+)]/index.svelte]: Too much for a page. It's better being split

File: src/components/ui/snackbar.svelte [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
-
-  let snackbar = null;
-  onMount(() => document.body.appendChild(snackbar));
-  onDestroy(() => document.body.removeChild(snackbar));
+  export function open() {
+    isOpen = true;
+    if (timeoutInstance != null) {
+      clearTimeout(timeoutInstance);
+    }
+    timeoutInstance = setTimeout(() => { isOpen = false; }, timeout);
```
Comment (aabounegm) [inline src/components/ui/snackbar.svelte]: Or simply
```suggestion
    timeoutInstance = setTimeout(close, timeout);
```

File: src/routes/product/[id([0-9]+)]/index.svelte (hunk 1 above)
Comment (illright) [reply src/routes/product/[id([0-9]+)]/index.svelte]: How would you suggest splitting it? I guess we could put the purchase modal in a container, but that's about it

File: src/components/ui/snackbar.svelte (hunk 2 above)
Comment (illright) [reply src/components/ui/snackbar.svelte]: Nah, the `close` function also involves clearing the timeout, and it's not particularly clear why would we wanna do that, so this one looks clearer to me

**Pull request merged**